### ISSUE TITLE
Typo in README.md[BUG]

### ISSUE THREAD

**Issue by johnhartman** [awslabs/aws-jwt-verify#78]
In the README.md file, line 309, 'customJwtChecks' should be 'customJwtCheck'.

**Comment by ottokruse** [comment]
Thanks! Also ... great opportunity to become a contributor to the repo :)

**Comment by dakshamdev** [comment]
> Thanks! Also ... great opportunity to become a contributor to the repo :)

A good one indeed :)

**Comment by ottokruse** [comment]
Thanks!

**Comment by johnhartman** [comment]
Hey Otto—

Quick question for you: for the multiple verifier, what order does the
verification occur in? Serially in order of the array declaration? Parallel
via Promise<result>?

best,
John
510.206.5846 (mobile)

**Comment by ottokruse** [comment]
Hi John. The JWT is just verified once actually. The array you pass in, is turned into a hash map of issuer configurations, where issuer is the key. During verification, using the issuer (`iss`) from the JWT payload, the right config is loaded from the hash map.

**Comment by johnhartman** [comment]
Thanks for the quick response. Was actually wondering about the order that
the tokens are decoded, or are they decoded in parallel? Here's the deal:
I'm on a project replacing a well-known identity provider with Cognito. New
API calls will handle the old provider's tokens if the tokens fail to
decode via cognito. Therefore I'd like to decode via cognito first, then
fall back to the old provider. Does the decode occur serially in the order
of the provider array, or in parallel via promises?

best,
John
510.206.5846 (mobile)

**Comment by ottokruse** [comment]
Hi John, if you use the multi-verifier from this library, your JWT will be decoded and verified once.

The lib will look at your JWT's `iss` claim. Cognito will put a different `iss` in the JWTs than your old provider. So by looking at the `kid` claim the library will know which __one__ of the issuer configurations (that you provided as an array) to use.

Does that make sense?

**Comment by johnhartman** [comment]
TL;DR: ignorant me :)

Didn't realize that parts 0 and 1 are not encrypted, just base64'ed. So was
thinking decrypting was required to determine who issued the token. All is
good now. Thank you for helping me.

best,
John
510.206.5846 (mobile)
